pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -276,6 +271,12 @@ def _reconstruct(cls, state: dict, sync_fn: Optional[Callable] = None) -> 'Resul
         result_metric.__setstate__(state, sync_fn=sync_fn)
         return result_metric
 
+    def to(self, *args: Any, **kwargs: Any) -> 'DeviceDtypeModuleMixin':
+        self.__dict__.update(
+            apply_to_collection(self.__dict__, (torch.Tensor, Metric), move_data_to_device, *args, **kwargs)
+        )
+        return self
```

Comment:
@SkafteNicki maybe this should also be done in the base `Metric` implementation, so `_forward_cache` is also moved to the correct device

Although we still need to override it here as we need to apply it to `Metric` types here too

cc @tchaton

Reply:
Yes, definitely.
Open an issue: https://github.com/PyTorchLightning/metrics/issues/392